src/mapport.cpp

```diff
@@ -153,44 +153,23 @@ void StartThreadMapPort()
 
 static void DispatchMapPort()
 {
-    if (g_mapport_current_proto == MapPortProtoFlag::NONE && g_mapport_enabled_protos == MapPortProtoFlag::NONE) {
-        return;
-    }
-
-    if (g_mapport_current_proto == MapPortProtoFlag::NONE && g_mapport_enabled_protos != MapPortProtoFlag::NONE) {
+    if (!g_mapport_current && g_mapport_enabled) {
```

Comment:
Maybe i'm missing something but the intent of `g_mapport_current` is hazy to me. It's turned on in `ThreadMapPort` while `ProcessPCP` is running, then turned off temporarily when `ProcessPCP` returns false.

- If we want to know if the mapport thread is running, `g_mapport_thread.joinable()` seems to suffice.
- If we want to know if mapport thread is supposed to be running, or stop it, there's `g_mapport_enabled`.

Reply:
Yes, as far as i can tell it's an artifact of switching between protocols. I should remove it. In fact i think `g_mapport_enabled` can be removed as well as we can just use `g_mapport_interrupt` to stop it.

Reply:
Done for both `current` and `enabled`.

Reply:
Nice! Good catch.